code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -386,6 +386,25 @@
 			return 2
 	return ..()
 
+///Runs the drone's inbuilt anti-grab measures, shocking the grabber after a while
+/mob/living/critter/flock/drone/proc/do_antigrab()
+	//if the only grab is a block then ignore it
+	if (!length(src.grabbed_by) || (length(src.grabbed_by) == 1 && src.find_type_in_hand(/obj/item/grab/block)))
+		src.antigrab_counter = 0
+		return
+
+	antigrab_counter++
+	if (antigrab_counter >= src.antigrab_fires_at)
+		playsound(src, "sound/effects/electric_shock.ogg", 40, 1, -3)
+		boutput(src, "<span class='flocksay'><b>\[SYSTEM: Anti-grapple countermeasures deployed.\]</b></span>")
+		for(var/obj/item/grab/G in src.grabbed_by)
+			if (istype(G, /obj/item/grab/block)) //do not shock ourselves
+				continue
```

Comment:
I don't think you can have a block up if grabbed?

Reply:
well then why are we checking length above :screm: